frontend/pages/project-developer/[id]/index.tsx

```diff
@@ -42,29 +47,42 @@ export const getServerSideProps = async ({ params: { id }, locale }) => {
     props: {
       intlMessages: await loadI18nMessages({ locale }),
       enums: groupBy(enums, 'type'),
-      // Fixing issues with circular references (caused by the inclusion of projects)
-      // https://github.com/vercel/next.js/discussions/10992#discussioncomment-59574
-      projectDeveloper: decycle(projectDeveloper),
+      initialProjectDeveloper: projectDeveloper,
     },
   };
 };
 
 type ProjectDeveloperPageProps = {
-  projectDeveloper: ProjectDeveloperType;
+  initialProjectDeveloper: ProjectDeveloperType;
   enums: GroupedEnumsType;
 };
 
 const ProjectDeveloperPage: PageComponent<ProjectDeveloperPageProps, StaticPageLayoutProps> = ({
-  projectDeveloper,
+  initialProjectDeveloper,
   enums,
 }) => {
+  const { projectDeveloper } = useProjectDeveloper(
+    initialProjectDeveloper.id,
+    {
+      includes: 'projects',
+    },
+    initialProjectDeveloper
+  );
+
+  const [isFavourite, setIsFavourite] = useState(projectDeveloper.favourite);
+
+  useEffect(() => {
+    // this useEffect is needed because the initial PD can be different from the current
```

Comment:
I was really confused about why this is still needed but I've finally understood. On the server, when we fetch the PD, we don't send the session cookie so the endpoint doesn't tell us if the PD is in the favourites. When the hook executes on the client (the browser), we do send the token and thus `projectDeveloper.favourite` has a different value.

Since you're storing the “favourite status” in the `useState` hook, you need to update it whenever the `useQuery` hook returns new data.

Another way you could have implemented this is if you wouldn't use `useState` at all and would invalidate the `useQuery` request when the mutation has succeeded.

I think this comment could be improved though ;).

Reply:
You are right, I will ad this explanation to the comment